### PR TITLE
`split` : `--filter` and stdin updates

### PR DESCRIPTION
**Pull request by zhitkoff**
Refactor handling of `--filter` option, specifically BrokenPipe error handling - for "endless" and finite inputs.
Better handling of stdin input errors for `--number` option.
Following GNU `split` behavior for above.

GNU test tests/split/filter.sh should stop failing

### CODE REVIEW

**Comment by zhitkoff** [comment]
@sylvestre would you mind reviewing this one?

File: src/uu/split/src/split.rs [hunk 1]
```diff
@@ -748,6 +748,9 @@ enum SettingsError {
     /// Multiple different separator characters
     MultipleSeparatorCharacters,
 
+    /// Using `--filter` with Kth chunk options that output to stdout
```
[the remark this answers is not in the document]
Comment (sylvestre) [reply src/uu/split/src/split.rs:751]: sorry, i mean in the comment :)

[the remark this answers is not in the document]
Comment (zhitkoff) [reply src/uu/split/src/split.rs]: got it, pushed an update

**Comment by sylvestre** [comment]
> GNU testsuite comparison:
> 
> ```
> Congrats! The gnu test tests/split/filter is no longer failing!
> ```

bravo :)

**Pull request merged**